### PR TITLE
Minor fixes in preparation for release v1.0

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
A number of things were changed in preparation for the initial release.
- Updated README and added release notes
- Updated docs to explain which versions of optimizers are tested, and added some installation instructions.
- Updated tests

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- Bugfix (non-breaking change which fixes an issue)
- Documentation update

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run unit and regression tests which pass locally with my changes
- [x] I have added necessary documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,17 +8,24 @@ An example output from OptView is shown below.
 
 ![Example](doc/OptView.png)
 
-Documentation
--------------
+## Optimizer Support
+pyOptSparse provides Python interfaces for a number of optimizers.
```
Comment (johnjasa) [inline README.md]: List the optimizers here

File: README.md [hunk 2]
```diff
@@ -9,17 +8,24 @@ An example output from OptView is shown below.
...
+However, we do not provide the source code for SNOPT and NLPQLP, due to their restrictive license requirements.
+Please contact the authors of the respective codes if you wish to obtain them.
```
Comment (johnjasa) [inline README.md]: Tell me where to find the authors of the codes? Are they within the files?

File: README.md [hunk 3]
```diff
@@ -9,17 +8,24 @@ An example output from OptView is shown below.
...
+Furthermore, ParOpt and IPOPT are available as a open source package but must be installed separately.
```
Comment (johnjasa) [inline README.md]: Link to the repos/sources for these optimizers

File: README.md [hunk 4]
```diff
@@ -9,17 +8,24 @@ An example output from OptView is shown below.
...
-Citation
---------
+## Testing
+Testing is done with the `testflo` package developed by the openMDAO team, which can be installed via `pip install testflo`.
+To run the tests, simply type `testflo .` in the root directory.
 
+## Citation
 Please cite pyOpt and the authors of the respective optimization
 algorithms in any publication for which you find it useful.
 For more background, theory, and figures, see the [pyOptSparse journal article](http://mdolab.engin.umich.edu/sites/default/files/pyOpt.pdf).
```
Comment (johnjasa) [inline README.md]: I want this to be more clear that pyOpt and pyOptSparse are separate products, and this is probably the best place to do that

File: doc/optimizers/paropt.rst [hunk 5]
```diff
@@ -5,6 +5,13 @@ ParOpt
 ParOpt is a nonlinear interior point optimizer that is designed for large parallel design optimization problems with structured sparse constraints.
 ParOpt is open source and can be downloaded at `https://github.com/gjkennedy/paropt <https://github.com/gjkennedy/paropt>`_.
 Documentation and examples for ParOpt can be found at `https://gjkennedy.github.io/paropt/ <https://gjkennedy.github.io/paropt/>`_.
+ParOpt does not provide version tagging, but the GitHub source code from September 2019 has been verified to work.
```
Comment (johnjasa) [inline doc/optimizers/paropt.rst]: Give a commit number to this

File: doc/optimizers/pyipopt.rst [hunk 6]
```diff
@@ -2,6 +2,15 @@
 
 IPOPT
 =====
+IPOPT (Interior Point OPTimizer) is an open source interior point optimizer, designed for large-scale nonlinear optimization.
+The source code can be found `here <https://github.com/coin-or/Ipopt>`_.
```
Comment (johnjasa) [inline doc/optimizers/pyipopt.rst]: Great! Nice to include where to get it.

File: README.md (hunk 3 above)
Comment (nwu63) [reply README.md]: Links are in the docs page for each optimizer.

File: README.md (hunk 2 above)
Comment (nwu63) [reply README.md]: Links in the docs page for each optimizer, but will add a line redirecting readers there.

**Pull request merged**